.github/workflows/changelog.yml

```
@@ -0,0 +1,115 @@
+# This workflow makes sure contributors don't forget to add a changelog entry or explicitly opt-out of it.
+
+name: Changelog
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - ready_for_review
+      - reopened
+      - synchronize
+      - labeled
+      - unlabeled
+
+# This workflow runs for not-yet-reviewed external contributions and so it
+# intentionally has no write access and only limited read access to the
+# repository.
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  check-changelog-entry:
+    name: "Check Changelog Entry"
+    runs-on: ubuntu-latest
+    concurrency:
+      group: changelog-${{ github.ref }}
```

pull_request_target sets the ref to the base branch (so e.g. main) so we need to use head_ref here

```suggestion
      group: changelog-${{ github.head_ref }}
```